packages/axelar-wasm-std/src/msg_id/base_58_solana_event_index.rs

```diff
@@ -0,0 +1,362 @@
+        assert!(res.is_err());
+
+        let res =
+            Base58SolanaTxDigestAndEventIndex::from_str(&format!("{}{}", tx_digest, event_index));
+        assert!(res.is_err());
+
+        for _ in 0..10 {
+            let random_sep: char = rand::random();
+            if random_sep == '-' {
+                continue;
+            }
+            let res = Base58SolanaTxDigestAndEventIndex::from_str(&format!(
+                "{}{}{}",
+                tx_digest, random_sep, event_index
+            ));
+            assert!(res.is_err());
+        }
+    }
+
+    #[test]
+    fn should_not_parse_msg_id_with_event_index_with_leading_zeroes() {
+        let tx_digest = random_tx_digest();
+        let res = Base58SolanaTxDigestAndEventIndex::from_str(&format!("{}-01", tx_digest));
+        assert!(res.is_err());
+    }
+
+    #[test]
+    fn should_not_parse_msg_id_with_non_integer_event_index() {
+        let tx_digest = random_tx_digest();
+        let res = Base58SolanaTxDigestAndEventIndex::from_str(&format!("{}-1.0", tx_digest));
+        assert!(res.is_err());
+
+        let res = Base58SolanaTxDigestAndEventIndex::from_str(&format!("{}-0x00", tx_digest));
+        assert!(res.is_err());
+
+        let res = Base58SolanaTxDigestAndEventIndex::from_str(&format!("{}-foobar", tx_digest));
+        assert!(res.is_err());
+
+        let res = Base58SolanaTxDigestAndEventIndex::from_str(&format!("{}-true", tx_digest));
+        assert!(res.is_err());
+
+        let res = Base58SolanaTxDigestAndEventIndex::from_str(&format!("{}-", tx_digest));
+        assert!(res.is_err());
+    }
+
+    #[test]
+    fn should_not_parse_msg_id_with_overflowing_event_index() {
+        let event_index: u64 = u64::MAX;
+        let tx_digest = random_tx_digest();
+        let res =
+            Base58SolanaTxDigestAndEventIndex::from_str(&format!("{}-{}", tx_digest, event_index));
+        assert!(res.is_err());
+    }
+
+    #[test]
+    fn trimming_leading_ones_should_change_bytes() {
+        for _ in 0..100 {
+            let mut bytes = random_bytes();
+
+            // set a random (non-zero) number of leading bytes to 0
+            let leading_zeroes = rand::random::<usize>() % bytes.len() + 1;
+            for b in bytes.iter_mut().take(leading_zeroes) {
+                *b = 0;
+            }
+
+            let b58 = bs58::encode(&bytes).into_string();
+
+            // verify the base58 has the expected number of leading 1's
+            for c in b58.chars().take(leading_zeroes) {
+                assert_eq!(c, '1');
+            }
+
+            // trim a random (non-zero) number of leading 1's
+            let trim = rand::random::<usize>() % leading_zeroes + 1;
+
+            // converting back to bytes should yield a different result
+            let decoded = bs58::decode(&b58[trim..]).into_vec().unwrap();
+            assert_ne!(bytes.to_vec(), decoded);
+        }
+    }
```

Comment:
I think this kind of tests are more "exploratory" ones of the `b58` library. Tests that provide insights about a library expected behaviour. If i am right, i would not put them here, but in a dedicated integration test suite, so not related with any specific part of the code.